app/controllers/accepts_controller.rb

```diff
@@ -5,8 +5,9 @@ def mail
     @user_email = @email
     hash = Digest::SHA512.hexdigest(@user_email.to_s)
     @str = (0...4).map { ('a'..'z').to_a[rand(26)] }.join
```

Comment:
I think `hash` is not used anymore. Remove it.

`str` is not a good name for the variable. Please rename it to clarify its intent.